.pre-commit-config.yaml

```diff
@@ -1,21 +1,68 @@
-repos:
-  # Using this mirror lets us use mypyc-compiled black, which is about 2x faster
-  # This is recommended by psf/black: https://github.com/psf/black/blob/main/.pre-commit-hooks.yaml
-  - repo: https://github.com/psf/black-pre-commit-mirror
-    rev: 23.7.0
-    hooks:
-      - id: black
-        # It is recommended to specify the latest version of Python
-        # supported by your project here, or alternatively use
-        # pre-commit's default_language_version, see
-        # https://pre-commit.com/#top_level-default_language_version
-        language_version: python3.10
-        files: ^(tests|schematic|schematic_api)/
-
-  - repo: https://github.com/pycqa/isort
-    rev: 5.13.2
-    hooks:
-      - id: isort
-        name: isort (python)
-        files: ^(tests|schematic|schematic_api)/
-        args: ["--profile", "black", "--filter-files"]
+exclude: "^docs/conf.py|^docs/build/"
+
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.4.0
+    hooks:
+      - id: trailing-whitespace
+      - id: check-added-large-files
+      - id: check-ast
+      # - id: check-json
+      - id: check-merge-conflict
+      - id: check-xml
+      - id: check-yaml
+        exclude: ^mkdocs\.yml$
+      - id: debug-statements
+      - id: end-of-file-fixer
+      - id: requirements-txt-fixer
+      - id: mixed-line-ending
+        args: ["--fix=auto"] # replace 'auto' with 'lf' to enforce Linux/Mac line endings or 'crlf' for Windows
+
+  # - repo: https://github.com/astral-sh/ruff-pre-commit
```

Comment:
Are the comments hooks we plan to turn on in the future? are we no longer using `isort` as well?

Reply:
^ so isort seems to cause a weird error with the code, so I turned it off for now. Will turn it back on later.

Reply:
that makes sense, sounds good

Reply:
Ya, we can certainly look into it!  My concern is always that `ruff` may clash against `black` or other formatting tools, but thats a concern i have about every formatting tool. 😓